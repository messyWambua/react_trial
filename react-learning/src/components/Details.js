import React, { Component } from "react";
import { connect } from "react-redux";
import AudioPlayer from 'material-ui-audio-player';
import equal from 'fast-deep-equal'

import { withStyles, withTheme, CircularProgress, Card, CardContent, Button, CardMedia, SkipNextIcon, Typography, CardActions, CardActionArea} from "../styles/materialUI"
import { fetchShow } from  "../store/actions/singleShowActions";
import { loadingAction } from "../store/actions/loadingActions";
import { convertSpeechToText } from "../store/actions/voiceActions";
import placeholderMovie from "../images/film-placeholder.jpg";
import stripHtml from "../utils/htmlToString";

const styles = () => ({
  card: {
    height: '100vh'
  },
  image: {
    position: 'relative',
        left: '50%', 
  },
  loader: {
    position: 'absolute',
    left: '50%', 
    top: '50%',
    transform: 'translate(-50%, -50%)',
  }
});

export class Show extends Component {
constructor(props) {
  super(props);
}



componentDidUpdate(prevProps) {
  if(!equal(this.props.show.id, prevProps.show.id) && Object.keys(this.props.show).length !== 0) {
      this.props.convertSpeechToText(this.props.show.summary);
  }
} 


 async componentDidMount() {
  const id = this.props.match.params.id;
  this.props.fetchShow(id);
}

renderLoading() {
  const  { classes } = this.props;
  return (
    <div>
      <CircularProgress isze="100" className={classes.loader} color="inherit" />
    </div>
  );
}

renderShows() {
  const { classes, speech, show } = this.props

  return (
    <div >
      <Card className={classes.card}>
      <CardActionArea>
        <img className={classes.image} src={show.image? show.image.medium : placeholderMovie } alt={show.name} />
        <CardContent>
        {<AudioPlayer
    elevation={1}
    width="100%"
    variation="default"
    spacing={3}
    download={true}
    order="standart"
    preload="auto"
    loop={true}
    src={speech}
  />}
          <Typography gutterBottom variant="h5" component="h2">
            {show.name}
          </Typography>
          <Typography variant="body2" color="textSecondary" component="p">
            {stripHtml(show.summary)}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Button size="small" color="default">
          Share
        </Button>
        <Button size="small" color="default">
          official site
        </Button>
      </CardActions>
    </Card>
    </div>
  );
}

render() {
 return(
  <div>
    {this.props.loading ? this.renderLoading() : this.renderShows()}
  </div>
 )
  
}
}

const mapStateToProps = state => ({
  show: state.show.show,
  loading: state.speech.loading,
  speech: state.speech.speech
});

export default connect(
  mapStateToProps,
  { fetchShow, loadingAction, convertSpeechToText }
  )(withStyles(styles)(Show));
