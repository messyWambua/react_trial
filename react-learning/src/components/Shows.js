import React, { Component } from "react";
import { connect } from "react-redux";

import { Link as RouterLink } from 'react-router-dom';
import { withStyles, GridList, GridListTile, GridListTileBar, CircularProgress} from "../styles/materialUI"
import { fetchShows } from  "../store/actions/allShowsActions";
import placeholderMovie from "../images/film-placeholder.jpg";


const Link1 = React.forwardRef((props, ref) => <RouterLink innerRef={ref} {...props} />);
const styles = theme => ({
  root: {
    display: 'flex',
    flexWrap: 'wrap',
    justifyContent: 'space-around',
    overflow: 'hidden',
    backgroundColor: theme.palette.background.paper,
  },
  gridList: {
    width: '100%',
    height: '100%',
  },
  zoom: {
    zIndex: 2,
    transition: 'transform .2s',
    transform: 'scale(1.2)'
  },
  loader: {
    position: 'absolute',
    left: '50%', 
    top: '50%',
    transform: 'translate(-50%, -50%)',
  }
});

export class Shows extends Component {
constructor(props) {
  super(props);
  this.state = {
    hover: false
  };
  this.hoverOff = this.hoverOff.bind(this);
}

hoverOn(id){
  this.setState({
    hover: id
  });
}

hoverOff(){
  this.setState({
    hover: false
  });
}

componentDidMount() {
  this.props.fetchShows();
}

renderLoading() {
  const  { classes } = this.props;
  return (
    <div className={classes.loader}>
      <CircularProgress color="inherit" size="100" data-test='spinner' />
    </div>
  );
}

renderShows() {
  const  { shows, classes } = this.props;
  return(
    <div className={classes.root}>
      <GridList cellHeight={400} cols={6} spacing={10} className={classes.gridList}>
        {shows.map(tile => (
          <GridListTile
          data-test='shows-cards-test'
          className={this.state.hover === tile.id ? classes.zoom : ""}
          onMouseEnter={this.hoverOn.bind(this, tile.id)}
          onMouseLeave={this.hoverOff}
          component={Link1}
          to={{ pathname: `/shows/${tile.show.id}`}} 
          key={tile.id}
          >
            <img src={tile.show.image? tile.show.image.medium : placeholderMovie } alt={tile.show.name} />
            <GridListTileBar
              title={tile.show.name}
              subtitle={<span>season: {tile.season}</span>}
            />
          </GridListTile>
        ))}
      </GridList>
    </div>
  );
}

render() {
  return (
    <div>
      {this.props.loading ? this.renderLoading() : this.renderShows()}
    </div>
  );
}
}

const mapStateToProps = state => ({
  shows: state.shows.shows,
  loading: state.shows.loading
});

export default connect(
  mapStateToProps, 
  { fetchShows }
  )(withStyles(styles)(Shows));
