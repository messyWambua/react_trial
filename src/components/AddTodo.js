import React, { Component } from "react";

export class AddTodo extends Component {
  state = {
    title: ""
  };

  typing = e => this.setState({ [e.target.name]: e.target.value });

  onSubmit = e => {
      e.preventDefault();
      this.props.addTodo(this.state.title);
      this.setState({title : ''});
  }
  render() {
    return (
      <div>
        <form onSubmit={this.onSubmit} style={{ display: "flex" }}>
          <input
            type="text"
            name="title"
            placeholder="Add todo"
            onChange={this.typing}
            value={this.state.title}
            style={{ flex: 10, padding: "5px 5px" }}
          />
          <input type="submit" placeholder="Submit" style={{ flex: 1 }} />
        </form>
      </div>
    );
  }
}

export default AddTodo;
