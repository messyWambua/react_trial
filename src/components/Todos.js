import React, { Component } from "react";
import PropTypes from "prop-types";

export class Todos extends Component {
  getStyle = () => {
    /*
            if(this.props.todo.completed){
                return {
                        textDecoration : 'line-through'
                    } 
                }
            else {
                return{
                    textDecoration : 'none'
                    }
                }
        */
    return {
      textDecoration: this.props.todo.completed ? "line-through" : "none"
    };
  };

  render() {
    const { id, title } = this.props.todo;
    return (
      <div style={this.getStyle()}>
        <p>
          <input type="checkbox" onChange={this.props.markIt.bind(this, id)} />{" "}
          {title}
          <button style={btnStyle} onClick={this.props.delTodo.bind(this, id)}>
            x
          </button>
        </p>
      </div>
    );
  }
}

Todos.protoTypes = {
  todo: PropTypes.object.isRequired
};
const btnStyle = {
  float: "right",
  padding: "5px 10px",
  color: "#fff",
  background: "#fff000",
  border: "none",
  borderRadius: "50%"
};

export default Todos;
