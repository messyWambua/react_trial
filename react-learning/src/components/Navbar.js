import React, { Component } from "react";
import { BrowserRouter as Router, Route, Switch, Link } from "react-router-dom";

import { AppBar, Toolbar, Typography, withStyles} from '../styles/materialUI'
import logo from '../images/logo.png'
import Shows from './Shows';
import Details from './Details';

const styles = () => ({
  logo: {
    maxWidth: '200px',
    maxHeight: '69px',
    display: 'block'
  },
  title: {
    flexGrow: 1,
    marginLeft: '50px',
  }
});

export class Navigation extends Component {
render() {
  const { classes } = this.props
  return (
    <Router>
      <AppBar position="sticky">
        <Toolbar>
        <Link to="/">
        <img src={logo} className={classes.logo} data-test="logo" />
        </Link>
        <Link to="/" style={{ textDecoration: 'none' }}>
        <Typography variant="h5" color="textPrimary" data-test="nav-links" className={classes.title}>
            Home
        </Typography>
        </Link>
        </Toolbar>
      </AppBar>
      <Switch>
        <Route path="/" exact component={Shows} />
        <Route path="/shows/:id" component={Details} />
        </ Switch>
    </Router>
  );
}
}


export default withStyles(styles)(Navigation);