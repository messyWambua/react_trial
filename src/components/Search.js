import React, { Component } from "react";
import Todos from "./Todos";
import PropTypes from "prop-types";

class Search extends Component {
  render() {
    console.log(this.props.searches);
    return this.props.searches.map(todo => (
      <Todos
        key={todo.id}
        todo={todo}
        markIt={this.props.markIt}
        delTodo={this.props.delTodo}
      />
    ));
  }
}

Search.protoTypes = {
  searches: PropTypes.array.isRequired
};

export default Search;
