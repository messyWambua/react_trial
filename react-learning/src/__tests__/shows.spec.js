import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

import { Shows } from "../components/Shows";

describe("<Shows /> component", () => {
  let props;
  let shallowWrapper;
  beforeEach(() => {
    props = {
      shows: [{
        id: 1747698,
        season: 2019,
        show: {
          id: 15779,
          name: "CBS News Sunday Morning",
          image: {
            medium: "http://static.tvmaze.com/uploads/images/medium_portrait/90/226497.jpg",
          }
        }
      }],
      classes: {
        root: 'app-root',
        gridList: 'grid',
        zoom: 'zoom',
        loader: 'loader'
      }
    };
    props.fetchShows = jest.fn();
    shallowWrapper = shallow(<Shows {...props} />);
  });

  test("translate on mouseover a show", () => {
    let hover = shallowWrapper.find("WithStyles(ForwardRef(GridListTile))");
    expect(shallowWrapper.state()).toEqual({ hover: false });
    hover.simulate("mouseenter");
    expect(shallowWrapper.state()).toEqual({ hover: 1747698 });
    hover.simulate("mouseleave");
    expect(shallowWrapper.state()).toEqual({ hover: false });
  });

  test("renders without crashing", () => {
    const wrapper = renderer.create(<BrowserRouter><Shows {...props} /></BrowserRouter>);
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  test("fetches shows on load", () => {
    const spy = jest.spyOn(shallowWrapper.instance().props, "fetchShows");
    expect(spy).toHaveBeenCalled();
  });

  test("show cards render correctly", () => {
    expect(shallowWrapper.find("[data-test='shows-cards-test']")).toHaveLength(
      1
    );
  });
});
