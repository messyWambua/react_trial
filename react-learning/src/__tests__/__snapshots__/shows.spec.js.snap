// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Shows /> component renders without crashing 1`] = `
<div>
  <div
    className="app-root"
  >
    <ul
      className="MuiGridList-root grid"
      style={
        Object {
          "margin": -5,
        }
      }
    >
      <a
        className="MuiGridListTile-root"
        data-test="shows-cards-test"
        href="/shows/15779"
        onClick={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        style={
          Object {
            "height": 410,
            "padding": 5,
            "width": "16.666666666666668%",
          }
        }
      >
        <div
          className="MuiGridListTile-tile"
        >
          <img
            alt="CBS News Sunday Morning"
            src="http://static.tvmaze.com/uploads/images/medium_portrait/90/226497.jpg"
          />
          <div
            className="MuiGridListTileBar-root MuiGridListTileBar-titlePositionBottom MuiGridListTileBar-rootSubtitle"
          >
            <div
              className="MuiGridListTileBar-titleWrap"
            >
              <div
                className="MuiGridListTileBar-title"
              >
                CBS News Sunday Morning
              </div>
              <div
                className="MuiGridListTileBar-subtitle"
              >
                <span>
                  season: 
                  2019
                </span>
              </div>
            </div>
          </div>
        </div>
      </a>
    </ul>
  </div>
</div>
`;
